### PR TITLE
Update docs readme with editing info

### PR DESCRIPTION
**Pull request by saltcod**
## What kind of change does this PR introduce?

Updates docs readme with instructions for how to contribute

### CODE REVIEW

File: apps/reference/README.md [hunk 1]
```diff
@@ -1,6 +1,6 @@
 # Reference Docs
 
-Supabase Reference Docs
+Home to all of the documentation for Supabase. Built with [https://docusaurus.io/](Docusaurus) and [https://typesense.org/](typesense)
```
Comment (MildTomato) [inline apps/reference/README.md:3]:
```suggestion
Documentation website for Supabase. Built with [https://docusaurus.io/](Docusaurus) and [https://typesense.org/](typesense)
```

File: apps/reference/README.md [hunk 2]
```diff
@@ -43,3 +43,30 @@ We use documentation specifications which can be used to generate human-readable
...
+
+- Branch from `master` and name your branches with the following structure
+  - `{type}/{branch_name}`
+    - Type: `chore | fix | feature`
+    - The branch name is arbitrary — just make sure it summarizes the work.
+- When you send a PR to `master`, tag [https://github.com/dannykng](@dannykng) for a review.
+
+### Which files to edit?
+
+Our docs site are comprised of guides, tutorials, quickstarts as well as reference docs. It's important to note that these reference docs (located at https://supabase.com/docs/reference) are generated from spec files. As the DocSpec section above notes, we use documentation specs to generate human-readable docs.
```
Comment (dannykng) [inline apps/reference/README.md:66]:
```suggestion
Our docs site are comprised of guides, tutorials, quickstarts, and reference docs. It's important to note that the [reference docs](https://supabase.com/docs/reference) are generated from spec files. As the DocSpec section above notes, we use documentation specs to generate human-readable docs.
```

File: apps/reference/README.md [hunk 3]
```diff
@@ -43,3 +43,30 @@ We use documentation specifications which can be used to generate human-readable
...
+
+Two quick examples to help clarify this:
+
+1. If you want to update the doc for the supabase-js [https://supabase.com/docs/reference/javascript/auth-signup](<`signUp()`>) function, you would need to edit that in the supabase-js [https://github.com/supabase/supabase/edit/master/spec/supabase_js_v1_legacy.yml](spec file). The files generated from these spec files are included in the repo, so it can be confusing to know which file to edit. Just make sure you're not editing a file in a `_generated` directory, nor editing a file with a `custom_edit_url` field in the frontmatter at the top of the doc.
```
Comment (dannykng) [inline apps/reference/README.md:70]: Might not need to mention the generated directory: https://github.com/supabase/supabase/pull/9078

File: apps/reference/README.md [hunk 4]
```diff
@@ -43,3 +43,30 @@ We use documentation specifications which can be used to generate human-readable
...
+
+## How to contribute?
+
+- Branch from `master` and name your branches with the following structure
+  - `{type}/{branch_name}`
+    - Type: `chore | fix | feature`
+    - The branch name is arbitrary — just make sure it summarizes the work.
+- When you send a PR to `master`, tag [https://github.com/dannykng](@dannykng) for a review.
+
+### Which files to edit?
```
Comment (dannykng) [inline apps/reference/README.md:64]: I don't think we're quite there yet, but the **Edit this page** button at the bottom of each doc should be the source of truth. We can circle back and add this once we have it all cleaned up.

File: apps/reference/README.md [hunk 5]
```diff
@@ -43,3 +43,30 @@ We use documentation specifications which can be used to generate human-readable
 
 The benefit of using custom specifications is that we can generate many other types from a strict schema (eg, HTML and manpages).
 It also means that we can switch any documentation system we want. On this site we use Docusaurus, but in Supabase's official website we use a custom React site and expose only a subset of the available API for each tool.
+
+## Local setup
+
+To run the docs locally, you need to:
```
Comment (dannykng) [inline apps/reference/README.md:49]: External contributors will have to fork the repo too.

**Review by dannykng — COMMENTED** [review]
Added a few thoughts. The examples are great!

File: apps/reference/README.md (hunk 3 above)
Comment (saltcod) [reply apps/reference/README.md:70]: @dannykng the new instruction then should be something like:  "some files are generated, etc.....the source of truth for which file to edit is in the "Edit this page" link at the bottom?

Comment (dannykng) [reply apps/reference/README.md:70]: Yep!

**Comment by saltcod** [comment]
Closing this one in favour of Jonny's :  https://github.com/supabase/supabase/pull/9102/files

**Pull request closed without merging**